python/activity_stream/activity_stream.py

```diff
@@ -150,6 +158,28 @@ def note_widget(self):
         """
         return self.ui.note_widget
 
+    def _get_pre_submit_callback(self):
+        """
+        The pre-submit callback. This is None if one is not set, or a Python
+        callable if it is. This callable is run prior to submission of a new
+        Note or Reply. Note that the first (and only) argument passed to the
+        callback will be the calling :class:`NoteInputWidget`.
+
+        :returns:   Python callable or None
+        """
+        return self._pre_submit_callback
+
+    def _set_pre_submit_callback(self, callback):
+        self._pre_submit_callback = callback
+        self.reply_dialog.note_widget.pre_submit_callback = callback
+        self.note_widget.pre_submit_callback = callback
+
+    pre_submit_callback = QtCore.Property(
```

Comment:
Also, do you know if the docs for the property get exposed when using this form?